### PR TITLE
BE: Column display switch (Task data display) [SCI-7442]

### PR DESCRIPTION
**Pull request by G-Chubinidze**
Jira ticket: [SCI-7442](https://scinote.atlassian.net/browse/SCI-7442)

### What was done
_backend for saving column visibility states for my_module table_

### CODE REVIEW

File: app/views/experiments/table.html.erb [hunk 1]
```diff
@@ -23,7 +23,7 @@
           </div>
         </div>
         <% Experiments::TableViewService::COLUMNS.each do |col| %>
-          <div class="table-header-cell <%= col %>-column">
+          <div class="table-header-cell <%= col %>-column <%= 'hidden' unless col.to_s.in?(@my_module_visible_table_columns) || col == :task_name %>">
```
Comment (aignatov-bio) [inline app/views/experiments/table.html.erb:26]: You just need to refresh with javascript column state.

**Review by aignatov-bio — CHANGES_REQUESTED** [review]

File: app/views/experiments/table.html.erb (hunk 1 above)
Comment (artoscinote) [reply app/views/experiments/table.html.erb:26]: Doesn't that already happens when you switch columns on and off? It seems to work OK, this is just the initial state of the table.

File: app/controllers/experiments_controller.rb [hunk 2]
```diff
@@ -96,6 +96,7 @@ def table
 
     @project = @experiment.project
     @active_modules = @experiment.my_modules.active.order(:name)
+    @my_module_visible_table_columns = current_user.settings['visible_my_module_table_columns'].presence || []
```
Comment (aignatov-bio) [inline app/controllers/experiments_controller.rb:99]: You meed to define default visible state, not empty array

Comment (G-Chubinidze) [reply app/controllers/experiments_controller.rb:99]: it has to be empty if all the columns are hidden, so the logic afterwards will hide all the columns in the backend too, only visible columns are added inside that array, if all the presence is nil, it just means that every column is hidden, not shown

Comment (aignatov-bio) [reply app/controllers/experiments_controller.rb:99]: So when the user first time comes to the table, will all columns be hidden?

Comment (G-Chubinidze) [reply app/controllers/experiments_controller.rb:99]: aha, i just checked and you're right, all columns are hidden, i will fix it

**Review by aignatov-bio — APPROVED** [review]

**Pull request merged**